src/Lecture4.hs

```diff
@@ -134,8 +144,25 @@ errors. We will simply return an optional result here.
 🕯 HINT: Use the 'readMaybe' function from the 'Text.Read' module.
 -}
 
+splitOn :: Char -> String -> [String]
+splitOn delim = uncurry (:) . foldr (\c (container, res) -> if c == delim 
+                  then ([], container : res)
+                  else (c : container, res)
+                ) ([], [])
+
+-- Question: Is there a way to create a PositiveInt type ?
+
 parseRow :: String -> Maybe Row
-parseRow = error "TODO"
+parseRow = validateValues . splitOn ','
+  where
+    validateValues ["", _, _] = Nothing
+    validateValues [f , s, t] = (readMaybe (dropSpaces s) :: Maybe TradeType) 
+                                  >>= \parsedTradeType -> case readMaybe (dropSpaces t) of
```

Comment:
You don't need to do `dropSpaces` and explicit type annotation here. `readMaybe` already drops all leading and trailing spaces, and due to type inference Haskell is able to guess the `TradeType` 🙂